tests/AbstractPdoEventStoreTest.php

```diff
@@ -486,6 +486,34 @@ public function it_doesnt_double_escape_metadata(): void
 
     /**
      * @test
+     * issue: https://github.com/prooph/pdo-event-store/issues/110
+     */
+    public function it_handles_invalid_json(): void
+    {
+        $this->expectException(JsonException::class);
+
+        $event = UserCreated::with(['name' => ['John', 'ßnow']], 1);
+        $event = $event->withAddedMetadata('key', 'value');
+
+        $streamName = new StreamName('Prooph\Model\User');
+        $stream = new Stream($streamName, new ArrayIterator([$event]), ['some' => ['metadata', 'as', 'well']]);
+
+        $this->eventStore->create($stream);
+
+        // Trigger an error when using an umlaut in the payload.
+        $this->connection->query('SET NAMES latin1');
+
+        $metadataMatcher = new MetadataMatcher();
+        $metadataMatcher = $metadataMatcher->withMetadataMatch(
+            'event_id', Operator::EQUALS(), $event->uuid()->toString(), FieldType::MESSAGE_PROPERTY()
+        );
+
+        $streamEvents = $this->eventStore->load($streamName, 1, null, $metadataMatcher);
+        $streamEvents->current(); // Trigger PdoStreamIterator.
```

Comment:
I am not sure if i used the loading correct. Is there a better way to trigger the PdoStreamIterator?

Reply:
that's good, problem is those tests for mariadb, and postgres are failing.

Reply:
damn. any idea to make the test more generic or move the test to MySqlEventStoreTest and write tests especially for mariadb etc?

Reply:
I don't know, I'm also good with some specific tests for each db type, at least we have it covered then.

Reply:
I have to play around a bit with mariadb and postgres (i am not working with them) but this will take some time. if anyone knows how to create an error while json_decode with mariadb/postgres it would speed up the things.
The next two weeks i'm still busy and can not promise i will find the time to get this done.
If you know mariadb/postgres it's probably a trifle to get the tests done.